### DISCUSSION TITLE
GUI?

### DISCUSSION THREAD

**Discussion by nirvdrum** [microsoft/gctoolkit#346] [Q&A] [answered]
I'm reading through "Optimizing Java: Practical Techniques for Improving JVM Application Performance" and it has a chapter on GC logging, monitoring, and tuning that pointed me at jClarity Censum. In the years since the book was published, jClarity was acquired by Microsoft. It's great to see the libraries open sourced, but I can't find the original GUI anywhere.

I'm looking to monitor the execution of an application running on TruffleRuby, which ships as part of the GraalVM suite of dynamic languages. Truffle languages are typically distributed in two ways, both of which support GC logging: as a JVM-based distribution and as a GraalVM Native Image compiled executable. I could maybe manually integrate gctoolkit in the JVM case, but it'd be quite cumbersome and would result in a distribution different from what ships with GraalVM. Integrating into the Native Image executable would be unworkable since it integrates some private code from Oracle.

Is there a GUI available anywhere to support monitoring cases where the application cannot be readily built and instrumented? If the Censum GUI is no longer available, has anyone built something like GCViewer using gctoolkit as the log processing engine?

**Comment by karianna** [comment, marked as answer]
Hi @nirvdrum - thanks for reaching out! I'm afraid at this stage we have no plans to release a GUI for GCToolkit.  Happy to be a sounding board if/when you have questions if you decide to build one!
